### ISSUE TITLE
Quote and pullquote blocks: consider to use a cite element for the citation instead of a footer

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#2647]
While the HTML 5.1 recommendation states that `<blockquote>` elements can have an optional citation "which must be within a footer or cite element", all the examples in the recommendation always use a `<cite>` element, whether it's wrapped in a footer or not. The only exceptions seems to be the "permalink" example. See https://www.w3.org/TR/html51/grouping-content.html#the-blockquote-element

Seems to me [the footer element](https://www.w3.org/TR/html51/sections.html#the-footer-element) is more suited for generic information about its section, while [the cite element](https://www.w3.org/TR/html51/textlevel-semantics.html#the-cite-element) is more specific to a quote.

For better semantics, I'd propose to simplify and just use a `<cite>` element.

Aside: when the nearest ancestor of a `<footer>` element is the body, then the footer is also a landmark region. This is  not the case in Gutenberg (and if it will be removed, it won't for sure) so there should be no concerns. Worth noting Firefox incorrectly reports all footer elements as landmarks whether they're in the scope of the body or not, but that's going to be fixed in Firefox 57, see: https://www.marcozehe.de/2017/08/30/firefox-57-will-less-chatty-screen-readers-situations/

In the screenshot below: footers in the demo content reported as landmarks in Firefox 55.0.3 (will be fixed in FF 57)

![screenshot 55](https://user-images.githubusercontent.com/1682452/29996234-251c659e-8ffb-11e7-84df-ce5c91fcef4c.png)

**Comment by mtias** [comment]
Done in #3665.
